src/battle/main_character_attack.cpp

```diff
@@ -0,0 +1,34 @@
+#include "main_character_attack.h"
+#include "../utils/utils.h"
+#include <iostream>
+
+void MainCharacterAttack(MainCharacter* main_character, Creature* target)
+{
+	int hit_main_character_roll =
+		utils::RollDice(1, 20) + (main_character->get_ability_score(0) - 10) / 2;
+	std::cout << "The main character rolled " << hit_main_character_roll << std::endl;
+
+	if (hit_main_character_roll > target->get_ac())
+	{
+		std::cout << "The main character has better roll than the AC of the "
+				  << target->get_creature_type() << ", main character hits!"
+				  << std::endl;
+
+		int damage_main_character_roll =
+			utils::RollDice(1, 10) + (main_character->get_ability_score(0) - 10) / 2;
```

Comment:
As a note for a future task, we should probably create a "ability_score_modifier" member array for Creature, just so we don't have these ugly looking lines.

Reply:
i agree